scipy/integrate/_quadrature.py

```diff
@@ -1055,46 +1055,53 @@ def newton_cotes(rn, equal=0):
 
     """
     try:
-        N = len(rn)-1
+        xp = array_namespace(rn)
+        N = rn.shape[0]-1
+        rn = xp_promote(rn, force_floating=True, xp=xp)
         if equal:
-            rn = np.arange(N+1)
-        elif np.all(np.diff(rn) == 1):
+            rn = xp.arange(N+1, dtype=rn.dtype, device=xp_device(rn))
+        elif not is_lazy_array(rn) and xp.all(xp.diff(rn) == 1):
             equal = 1
     except Exception:
+        xp = np
         N = rn
-        rn = np.arange(N+1)
+        rn = xp.arange(N+1.)
         equal = 1
 
+    dtype = rn.dtype
+    device = xp_device(rn)
+
     if equal and N in _builtincoeffs:
         na, da, vi, nb, db = _builtincoeffs[N]
-        an = na * np.array(vi, dtype=float) / da
-        return an, float(nb)/db
```

Comment:
Previously, the second output of this function was a Python `float` here but a NumPy scalar below. We'll change the function to consistently return a NumPy scalar or 0d array of the chosen backend.